### ISSUE TITLE
ComboBox with setAllowCustomValue(true) and setValue("Unknown") throws Exception

### ISSUE THREAD

**Issue by knoobie** [vaadin/vaadin-combo-box#946]
If I have a ComboBox and setAllowCustomValue to true, I would expect that I can set an "unknown" value from the server-side as well. The JavaDoc does say the following:

     * If {@code true}, the user can input a value that is not present in the
     * items list. {@code value} property will be set to the input value in this
     * case. Also, when {@code value} is set programmatically, the input value
     * will be set to reflect that value.

Instead of setting the value to my unknown value, I get an exception:

java.lang.IllegalArgumentException: The provided value is not part of ComboBox: UNKNWON
	at com.vaadin.flow.component.combobox.ComboBox.setValue(ComboBox.java:526)
	at com.vaadin.flow.data.binder.Binder$BindingImpl.initFieldValue(Binder.java:1119)
	at com.vaadin.flow.data.binder.Binder$BindingImpl.access$200(Binder.java:966)
	at com.vaadin.flow.data.binder.Binder.lambda$setBean$1(Binder.java:1637)

Usecase: 

I would like to create a readOnly implementation of an EditView and I don't want to load all the items from 10 Comboboxes. I just want to show the current value in a readOnly way. No changes from the user allowed. So no other data required.

**Comment by web-padawan** [comment]
Closing in favor of #940. See also the commend mentioning that this is fixed in Vaadin 14: https://github.com/vaadin/vaadin-combo-box/issues/940#issuecomment-492217142

**Comment by knoobie** [comment]
I can confirm that it is fixed in the latest version `14.4.0`.

Example:

```java
    ComboBox<String> field = new ComboBox<>("Test");
    field.setAllowCustomValue(true);
    field.setItems(new ArrayList<>());
    field.setValue("UNKNOWN");
    field.setReadOnly(true);
```
